invenio_app_ils/permissions.py

```diff
@@ -89,6 +97,73 @@ def __init__(self, record):
         )
 
 
+class LoanExtendPermission(Permission):
+    """Return Permission to evaluate if the user can extend the loan."""
+
+    def __init__(self, record):
+        """Constructor."""
+        self.record = record
+
+        # NOTE: extra validation if the user is not librarian or admin
+        if current_user.id == int(record["patron_pid"]):
+            self.validate()
+
+        super().__init__(
+            UserNeed(int(record["patron_pid"])), backoffice_access_action
+        )
+
+    def validate(self):
+        """Validate if the loan can be extended."""
+        if not self.validate_extend_enabled():
+            raise InvalidLoanExtendError(
+                "You can extend this loan {} days before it "
+                "expires!".format(
+                    current_app.config["ILS_LOAN_WILL_EXPIRE_DAYS"])
+            )
+
+        if not self.validate_max_extensions():
+            raise InvalidLoanExtendError(
+                "You have reached the max number of extensions for THIS loan!"
```

Comment:
why `THIS` uppercase? :)

Reply:
I guess you wanted to stand out, on your comment your wrote `<THIS>` will make it right case :)